cinema/views.py

```diff
@@ -34,6 +38,29 @@ class MovieViewSet(viewsets.ModelViewSet):
     queryset = Movie.objects.all()
     serializer_class = MovieSerializer
 
+    def get_queryset(self):
+        queryset = Movie.objects.all()
+
+        if self.action in ("list", "retrieve"):
+            queryset = queryset.prefetch_related("genres", "actors")
+
+        genres = self.request.query_params.get("genres")
+        actors = self.request.query_params.get("actors")
+        title = self.request.query_params.get("title")
+
+        if genres:
+            genres = [genre for genre in genres.split(",")]
+            queryset = queryset.filter(genres__in=genres)
+
+        if actors:
+            actors = [actor for actor in actors.split(",")]
+            queryset = queryset.filter(actors__in=actors)
+
+        if title:
+            queryset = queryset.filter(title__icontains=title)
```

Comment:
Just a recommendation. You can use a walrus operator here. It can reduce number of code lines.